omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/DiagnosesFragmentController.java

```diff
@@ -85,24 +100,17 @@
         return ret;
     }
 
+    @Deprecated
     public List<SimpleObject> searchNonCoded(UiSessionContext context,
                                      UiUtils ui,
                                      @SpringBean("emrApiProperties") EmrApiProperties emrApiProperties,
                                      @SpringBean("emrConceptService") EmrConceptService emrConceptService,
+                                     @SpringBean("conceptService") ConceptService conceptService,
                                      @RequestParam("term") String query,
                                      @RequestParam(value = "start", defaultValue = "0") Integer start,
                                      @RequestParam(value = "size", defaultValue = "50") Integer size) throws Exception {
 
-        Collection<Concept> diagnosisSets = emrApiProperties.getDiagnosisSets();
-        Locale locale = context.getLocale();
-
-        List<ConceptSource> sources = emrApiProperties.getConceptSourcesForDiagnosisSearch();
-
-        List<ConceptSearchResult> hits = emrConceptService.conceptSearch(query, locale, null, diagnosisSets, sources, null);
-        List<SimpleObject> ret = new ArrayList<SimpleObject>();
-        for (ConceptSearchResult hit : hits) {
-            ret.add(simplify(hit, ui, locale));
-        }
+        List<SimpleObject> ret = search(context, ui, emrApiProperties, emrConceptService, conceptService, query, null, null, start, size);
```

Comment:
Ah thanks for this. I would return directly:
```java
return search(context, ui, emrApiProperties, emrConceptService, conceptService, query, null, null, start, size);
```
So we leave absolutely zero logic in the deprecated function.

Reply:
Correct.